### PR TITLE
Option to display id instead of name in state_ouptut: mixed

### LINKED ISSUE #35137: Option for salt on master returning ID instead of Name?
### Description of Issue/Question
Currently, when applying states to minions, the salt command will return a list of "Names" with their respective results. For example:

    Name: irhan - Function: user.present - Result: ...
    Name: irhan - Function: group.present - Result: ... 
    Name: irhan - Function: user.present - Result: ...
    Name: shadow.set_password - Function: module.run - Result: ...
    Name: /home/irhan/my_password - Function: file.managed - Result: ...
    Name: ssh-rsa AAAAB3NzaC1yc2EAAAABJQAAAQEAnRT1pOmVLYbN10R2237f/y6SWffn4JjJsInu0C
    z17YrBQFtEgZL0Lua6BB6RaSwiX3+3qLrh+vGJZgW/nVav9BUjWiXSePcBWPY9YJwrp0fd7mjcJDtfY7
    4vfLKDCVREyVEZLxwzPCui035GHJ/KWx1RA3LX7q0USfEVr25lgb0A7Pyj52xd7yqxsLlHkXvgOrjB9r
    XmhxpRZlYWlLpxKtbBXYI6b5fYhrfheDYoES9C6bjshAL6GWUX0FO06g861PuDUU6NLm/DO7lizSxNmw
    yoYwbZE9kYSAWnLGTTfcRncynEl4sf7nw8FBDMGUgKqUG3tf2+Hl1TuYkgBf5tCQ== irhan@key_01 
    - Function: ssh_auth.present - Result: ...

(As you can see, this is especially annoying for states which takes a _very_ long name, such as the `ssh_auth.present` state; its "Name" attribute is simply a mess of hundreds of characters...)

Inside the `.sls` file, though I have taken the pains of marking each state with a unique, descriptive state. The above lines, for instance, are given the following IDs:

    irhan_account:
    irhan_details:
    irhan_key:
    irhan_passwd:

Is it possible for the `salt` command to return IDs instead of Names?

The above output then would be (respectively):

    ID: irhan_account - Function: user.present - Result: ...
    ID: irhan_details - Function: group.present - Result: ... 
    ID: irhan_details - Function: user.present - Result: ...
    ID: irhan_passwd - Function: module.run - Result: ...
    ID: irhan_passwd - Function: file.managed - Result: ...
    ID: irhan_key - Function: ssh_auth.present - Result: ...

Tidy and nice :wink: 

### Versions Report
    Salt Version:
               Salt: 2016.3.1
    
    Dependency Versions:
               cffi: 1.7.0
           cherrypy: Not Installed
           dateutil: 1.5
              gitdb: 0.5.4
          gitpython: 0.3.2 RC1
              ioflo: Not Installed
             Jinja2: 2.7.2
            libgit2: Not Installed
            libnacl: Not Installed
           M2Crypto: 0.21.1
               Mako: 0.9.1
       msgpack-pure: Not Installed
     msgpack-python: 0.3.0
       mysql-python: 1.2.3
          pycparser: 2.14
           pycrypto: 2.6.1
             pygit2: Not Installed
             Python: 2.7.6 (default, Jun 22 2015, 17:58:13)
       python-gnupg: Not Installed
             PyYAML: 3.11
              PyZMQ: 14.0.1
               RAET: Not Installed
              smmap: 0.8.2
            timelib: Not Installed
            Tornado: 4.3
                ZMQ: 4.0.5
    
    System Versions:
               dist: Ubuntu 14.04 trusty
            machine: x86_64
            release: 3.13.0-92-generic
             system: Linux
            version: Ubuntu 14.04 trusty

### PR DESCRIPTION
**Pull request by cachedout**
Closes #35137

### CODE REVIEW

**Comment by rallytime** [comment]
@cachedout Some docs for this option would be good. Maybe on this page: https://docs.saltstack.com/en/latest/ref/output/all/salt.output.highstate.html and the Carbon release notes?

**Comment by cachedout** [comment]
@rallytime Thanks. Done.

File: salt/output/highstate.py [hunk 1]
```diff
@@ -16,15 +16,18 @@
...
+    ``mixed``, ``mixed_id``, ``changes`` and ``filter``.
 
     * The default is set to ``full``, which will display many lines of detailed
       information for each executed chunk.
     * If ``terse`` is used, then the output is greatly simplified and shown in
       only one line.
     * If ``mixed`` is used, then terse output will be used unless a state
       failed, in which case full output will be used.
+    * If ``mixed_id`` is used, then the mixed form will be used, but the
+      be drawn from the state ID. This is useful for cases where the name
```
Comment (rallytime) [inline salt/output/highstate.py]: `but the ____ will be drawn`?

**Comment by pepoluan** [comment]
Oooh, missed this! Let me test...

**Pull request merged**